### ISSUE TITLE
Constantly registering backend for host

### ISSUE THREAD

**Issue by haswalt** [rancher/rancher#1663]
Log files seem to be getting filled with:

```
time="2015-07-27T18:14:42Z" level=info msg="Registering backend for host [6122445f-1659-4b27-95a3-62b551810079]"
```

Is this correct? Seems odd that a functional and connected host would need constantly registering. If this is correct, logging this repeatedly seems a waste of log space

**Comment by cjellick** [comment]
Hi @haswalt 

TL;DR: If you have a proxy such as nginx configured to sit in front cattle, this is a known issue. We fixed it and it will be in the next build of rancher/server that we promote to be "latest."

Longer explanation:
The backend host should not need to constantly re-gesister. The host should have to do that once on startup of rancher-agent and after that only if there is a disruption in the connection between the host and the microservice that it is connecting to.

Do you have a proxy (like nginx) sitting in front of cattle? If so, here is what is likely happening:
* When rancher-agent starts up, it makes a long-lived websocket connection to a microservice we call the *websocket-proxy**. When you put  a proxy like nginx in front of cattle, it is actually sitting in front of this microservice.
* If your configured proxy kills idle connections, the host has to reconnect. In nginx's case, there is a timeout that kills idle long-lived connections. The setting is `proxy_read_timeout` and by default it is set to 60 seconds. So, if this connection is idle for 60 seconds, nginx kills it and the host reconnects, causing that log message.

The bug was that we weren't sending any websocket pings to keep the connection alive and the fix was to start sending those. Relevant PR:
https://github.com/rancher/websocket-proxy/pull/15



\* *This micro-services is a proxy that  that lets the UI pull stats and container logs from a host without having to connect directly to the host.*

**Comment by haswalt** [comment]
@cjellick Thanks for the explanation. I've got harpy sitting in front of the server. quick question then:

Can I get the agents to use the private network for the websocket instead of the public ip?

Otherwise i'll try setting the connection timeout to a huge number in haproxy

**Comment by cjellick** [comment]
@haswalt Yes, you can configure the hosts to connect over the private network. Just set that value in the "Host Registration" dialog in the UI. It can be accessed through the drop down near the top right of the UI. 

Once you change that setting, rancher will create a new `docker run ..rancher/agent` command using the new URL. If you want ***all**** traffic from host to the rest of rancher to be over that private IP, you would need to rerun the `docker run ... rancher/agent` command on each host. If you are just concerned with the long-lived  websocket-proxy connections, just restarting the rancher-agent container should work.

Finally, FWIW, we fixed some bugs related to the Host Registration setting and https last week. They'll land in latest the same time as the ping-fix. So, if you're going to configure that setting to be https, its best to hold off until our next release. But, since you said "private IP", you must not be looking to set that up over https, so you should be good to go.

\* The long-lived websocket connection is not the only outbound request to Rancher made by rancher-agent. Rancher-agent also makes normal http GETs/POSTs/PUTs to rancher-server to update the state of resources.

**Comment by haswalt** [comment]
Thanks again @cjellick I'll close this for now as a configuration issue. Any ideas on the release schedule?
